src/plugins/intel_cpu/src/emitters/x64/jit_emitter.hpp

```diff
@@ -130,6 +132,74 @@ class jit_emitter : public ov::snippets::Emitter {
         }
     }
 
+    // below 4 functions must be inline funtions to avoid corrupted rsp by function call, so defined inside class declaration.
```

Comment:
a function call itself will push rsp before function body execution and pop rsp after execution, such as save current code address, etc.  For example, if non-inline function is used, rsp in `internal_call_rsp_align` is 16-byte aligned, but after pop when finish `internal_call_rsp_align`, rsp maybe not aligned again.

Reply:
1) Yes, function call change rsp register. However, we should not care too much about it since this function call is performed during compilation stage and h->rsp is not affected. In other words, no instructions that change h->rsp are emitted.
2) We need to worry about how a function call would affect rsp only if we call this function in runtime (but why would we do that?). Then yes, we need to avoid call instruction (since it will change rsp) and opt for inline execution. However, since we compile the kernel ourselves, the inline execution means we need to emit relevant code during compilation stage (see 1). This has nothing to do with inline modifier or implementation placement, since h->rsp modification is controlled by us, not by a C++ compiler directly.

So could you please elaborate on how does implementation placement affect h->rsp modification?

Reply:
discussed offline, not force inline is needed, and implementation is moved to cpp